benchexec/tablegenerator/react-table/src/components/ReactTable.js

```diff
@@ -645,13 +647,20 @@ const Table = (props) => {
 
       newFilteredColumnValues[runset] = currentRunsetFilters;
     }
+    return newFilteredColumnValues;
+  }
+
+  // get selected status and category values
+  useEffect(() => {
+    let newFilteredColumnValues = getNewFileteredColumnValues();
     if (!deepEqual(newFilteredColumnValues, filteredColumnValues)) {
       setFilteredColumnValues(newFilteredColumnValues);
     }
     // Set current page to new last page if current page is not legit any more after filtering
     if (pageIndex >= pageCount) {
       gotoPage(pageCount - 1);
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Could you please explain why we need to disable this warning? Can we fix the warning instead?

It is always good to leave a comment describing why the warning is not relevant.

Reply:
react-hooks/exhaustive-deps shows the warning:  `useEffect` has a missing dependency: `getNewFilteredColumnValues`. 
But `getNewFilteredColumnValues` never changes and `useEffect` has the same working without it as a dependency.

If it needs to be included, then `getNewFilteredColumnValues` needs to be converted to callback function as it may cause unnecessary renders, and then we can't use the `getNewFilteredColumnValues` callback function to assign `filteredColumnValues` as it gives another warning: `getNewFilteredColumnValues` was used before it was defined. 
So ignoring the first warning is more preferable I guess.

Reply:
Thanks! This is a really useful explanation.